src/chart/rendering/drawingcontext.h

```diff
@@ -20,31 +20,50 @@ class DrawingContext
 	DrawingContext(Gfx::ICanvas &canvas,
 	    const Layout &layout,
 	    const Events &events,
-	    const Gen::Plot &plot,
-	    const CoordinateSystem &coordSys,
+	    const std::shared_ptr<Gen::Plot> &plot,
+	    const Styles::Chart &styles,
 	    RenderedChart &renderedChart) :
 	    plot(plot),
-	    coordSys(coordSys),
+	    coordSys(renderedChart.getCoordSys()),
 	    canvas(canvas),
 	    painter(*static_cast<Painter *>(canvas.getPainter())),
-	    options(*plot.getOptions()),
-	    rootStyle(plot.getStyle()),
+	    rootStyle(styles),
 	    rootEvents(events),
 	    layout(layout),
 	    renderedChart(renderedChart)
 	{
+		if (plot) {
```

Comment:
Why is this better to be inlined here than having it in the previous place, in a dedicated method getCoordSystem()?

Reply:
because there was no cached coordSystem, and sometimes it was a faulty usage (eg: when coordinate transformation was queried from the API).
In a later version, this will be moved back to Chart::draw, and `CoordinateSystem` default arguments will be filled in the constructor.
![kép](https://github.com/vizzuhq/vizzu-lib/assets/6457941/d72566b2-8dc0-4980-91c2-d1dc394ae658)